packages/integration-sdk-cli/src/generator/entitiesFlow.ts

```diff
@@ -0,0 +1,190 @@
+import { askAgain } from './util';
+import { snakeCase } from 'lodash';
+
+export type Entity = {
+  resourceName: string;
+  _type: string;
+  _class: string[];
+};
+
+async function entitiesFlow(inquirer, vendorName): Promise<Entity[]> {
+  const entities: Entity[] = [];
+  const useEntityFlow = await inquirer.prompt({
+    type: 'confirm',
+    name: 'value',
+    message: 'Do you want to declare entities?',
+  });
+  if (!useEntityFlow.value) return entities;
+
+  let again = false;
+  do {
+    const entity = await createEntityPrompt(inquirer, vendorName);
+    entities.push(entity);
+    again = await askAgain(inquirer);
+  } while (again);
+  return entities;
+}
+
+async function createEntityPrompt(inquirer, vendorName) {
+  const { resourceName } = await inquirer.prompt({
+    type: 'input',
+    name: 'resourceName',
+    message:
+      'What is the name of the entity? ex. User, EC2 Instance, Access Key',
+  });
+
+  const { _class } = await inquirer.prompt({
+    type: 'checkbox-plus',
+    name: '_class',
+    message: 'Which classes should the entity have?',
+    searchable: true,
+    pageSize: 10,
+    validate(input) {
+      if (input?.length > 0) {
+        return true;
+      } else {
+        // TODO: It would be nice to use an error message here
```

Comment:
Looks like this may be supported here: https://github.com/SBoudrias/Inquirer.js/tree/master/packages/inquirer#user-interfaces-and-layouts